dataland-internal-storage/src/main/kotlin/org/dataland/datalandinternalstorage/services/DatabaseDataStore.kt

```diff
@@ -1,28 +1,62 @@
 package org.dataland.datalandinternalstorage.services
 
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.dataland.datalandbackend.openApiClient.api.TemporarilyCachedDataControllerApi
 import org.dataland.datalandinternalstorage.entities.DataItem
 import org.dataland.datalandinternalstorage.repositories.DataItemRepository
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.dataland.datalandmessagequeueutils.constants.MqConstants
+import org.slf4j.LoggerFactory
+import org.springframework.amqp.core.Message
+import org.springframework.amqp.rabbit.annotation.Exchange
+import org.springframework.amqp.rabbit.annotation.Queue
+import org.springframework.amqp.rabbit.annotation.QueueBinding
+import org.springframework.amqp.rabbit.annotation.RabbitListener
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
-import java.util.UUID
 
 /**
  * Simple implementation of a data store using a postgres database
+ * @param dataItemRepository
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
+ * @param temporarilyCachedDataClient the service for retrieving data from the temporary storage
+ * @param objectMapper object mapper used for converting data classes to strings and vice versa
  */
 @Component
 class DatabaseDataStore(
     @Autowired private var dataItemRepository: DataItemRepository,
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired var temporarilyCachedDataClient: TemporarilyCachedDataControllerApi,
+    @Autowired var objectMapper: ObjectMapper,
 ) {
+    private val logger = LoggerFactory.getLogger(javaClass)
 
     /**
-     * Insterts data into a database
-     * @param data a json object
-     * @return id associated with the stored data
+     * Method that listens to the storage_queue and stores data into the database in case there is a message on the
+     * storage_queue
+     * @param message Message retrieved from storage_queue
      */
-    fun insertDataSet(data: String): String {
-        val dataID = "${UUID.randomUUID()}:${UUID.randomUUID()}_${UUID.randomUUID()}"
-        dataItemRepository.save(DataItem(dataID, data))
-        return dataID
+
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value = Queue("dataReceivedInternalStorageDatabaseDataStore"),
+                exchange = Exchange(MqConstants.dataReceived, declare = "false"),
+                key = [""],
+            ),
+        ],
+    )
+    fun persistentlyStoreDataAndSendMessage(message: Message) {
+        val dataId = cloudEventMessageHandler.bodyToString(message)
+        val correlationId = message.messageProperties.headers["cloudEvents:id"].toString()
```

Comment:
I've see those two rows (at least) twice. Should be a common method.

Reply:
Shifted them into the parameters

Reply:
ok